packages/iTwinUI-react/src/core/Table/Table.tsx

```diff
@@ -414,36 +414,32 @@ export const Table = <
     return flatColumns.some((column) => column.id === SELECTION_CELL_ID);
   }, [columns]);
 
-  const disableUserSelect = React.useCallback(
-    (e: KeyboardEvent) => {
-      if (e.key === 'Shift') {
-        ownerDocument &&
-          (ownerDocument.documentElement.style.userSelect = 'none');
-      }
-    },
-    [ownerDocument],
-  );
+  const disableUserSelect = React.useCallback((e: KeyboardEvent) => {
+    if (e.key === 'Shift') {
+      ownerDocument.current &&
+        (ownerDocument.current.documentElement.style.userSelect = 'none');
+    }
+  }, []);
```

Comment:
When using `useState`, its value can be changed with setter function, therefore it has to be in the deps list to prevent calling function in `useCallback` with an old value. When using `useRef`, value never changes as it is an object whose reference is always the same, only `value.current` changes. Therefore it is not needed in deps list since reference never changes and `value.current` inside `useCallback` function will take new values whenever it is called.